src/gui/dialogs/title_screen.cpp

```diff
@@ -207,23 +209,21 @@ void title_screen::init_callbacks()
 			//
 			// Use pango markup to insert drop cap
 			//
-			if (tip.text().str().substr(0,1) != "<") {
-				// Example: Lawful units -> <span ...>L</span>awful units
-				widget["label"] = "<span font_family='OldaniaADFStd' font_size='xx-large'>" + tip.text().str().substr(0,1) + "</span>" + tip.text().str().substr(1);
-			} else {
-				// Tip starts with a tag, so we need to insert the <span> after it
-				// then insert the </span> tag after the first character of the text
-				// after markup. Example:
-				// <i>Lawful</i> units -> <i><span ...>L</span>awful</i> units
-				std::size_t id = tip.text().str().find_first_of(">") + 1;
-				std::stringstream tip_text;
-				tip_text << tip.text().str().substr(0,id) // existing start tag
-						 << "<span font_family='OldaniaADFStd' font_size='xx-large'>" // span start tag
-						 << tip.text().str().substr(id,1) // first character
-						 << "</span>" // span end tag
-						 << tip.text().str().substr(id+1); // rest of the text
-				widget["label"] = tip_text.str();
+			// Example: Lawful units -> <span ...>L</span>awful units
+			// If tip starts with a tag, we need to insert the <span> after it
+			// then insert the </span> tag after the first character of the text
+			// after markup.
+			// Example: <i>Lawful</i> units -> <i><span ...>L</span>awful</i> units
+			const std::string& script_font = font::get_font_families(font::FONT_SCRIPT);
+			std::string tip_text = tip.text().str();
+			std::size_t pos = 0;
+			while (tip_text.at(pos) == '<' && pos < utf8::size(tip_text)) {
+				pos = tip_text.find_first_of(">", pos) + 1;
 			}
+			utf8::insert(tip_text, pos+1, "</span>");
+			utf8::insert(tip_text, pos, "<span font_family='" + script_font + "' font_size='xx-large'>");
```

Comment:
>  Is it guaranteed that pos is the right character count in the tip-starts-with-a-tag case?

I tested it 1. when the tip text starts with a tag, 2. when it doesn't and it seems to work correctly. Do you mean something else? Do you mean when there's a unicode character inside the '<' and '>' (e.g. <ব>)?

> script_font can be a comma separated list of fonts, right? Does that work here?

Yes, that seems to work. Pango tries the fonts until it finds the one that exists, from left to right, and uses it.

Reply:
> > Is it guaranteed that pos is the right character count in the tip-starts-with-a-tag case?
> 
> I tested it in the cases: 1. when the tip text starts with one or more tags, 2. when it doesn't and it seems to work correctly. Do you mean something else? Do you mean when there's a unicode character inside the '<' and '>' (e.g. <ব>)?

When there are multi-byte characters involved, yes. `pos` is calculated as a byte offset. Unless there are only single-byte characters involved you can't use it as a character position.

Either we can guarantee that the markup only involves single-byte characters (and add a note to that effect) or we could add byte offset support to `utf8::index` and maybe `utf8::insert` and use that.

Reply:
> Either we can guarantee that the markup only involves single-byte characters (and add a note to that effect)

Since this is pango markup, I think we can guarantee that. I'd stick with the first approach for now and add a comment. Best not to overcomplicate things.

Reply:
See a34c7deb46640698fa25ce601a977900c1699b37.

Reply:
There appears to be no issue with `<span font_family='fönt'>` for example but it's unlikely to happen and the assumption is documented so it's fine with me.